Upgrade TargetSdk, Kotlin, KSP, Gradle and Compose

pushpalroymm opened the pull request:
- Bump TargetSdk to `33` 
- Bump Kotlin to `1.8.10`
- Bump  KSP to `1.8.10-1.0.9`
- Bump  Gradle to `7.6`
- Bump AGP to `7.4.1`
- Bump Compose Compiler to `1.4.2`
- Add Compose BOM `2023.01.00`
- Replace deprecated code

pushpalroymm committed:
Bump targetSdk, kotlin, ksp, gradle, apg and others.

shubhamsinghmutualmobile commented on app/build.gradle:
```diff
@@ -1,16 +1,16 @@
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
-    id 'com.google.devtools.ksp' version "1.6.10-1.0.4"
+    id 'com.google.devtools.ksp' version "1.7.21-1.0.8"
```
Can we go to the latest stable version of both Kotlin and KSP? For Kotlin, it's `1.8.10` and for KSP, it's `1.8.10-1.0.9`

shubhamsinghmutualmobile commented on build.gradle:
```diff
@@ -1,12 +1,12 @@
-    id 'com.android.application' version '7.1.2' apply false
-    id 'com.android.library' version '7.1.2' apply false
-    id 'org.jetbrains.kotlin.android' version '1.6.10' apply false
+    id 'com.android.application' version '7.4.0' apply false
+    id 'com.android.library' version '7.4.0' apply false
```
Latest version is `7.4.1`

shubhamsinghmutualmobile commented on build.gradle:
```diff
@@ -1,12 +1,12 @@
-    id 'org.jetbrains.kotlin.android' version '1.6.10' apply false
+    id 'com.android.application' version '7.4.0' apply false
+    id 'com.android.library' version '7.4.0' apply false
+    id 'org.jetbrains.kotlin.android' version '1.7.21' apply false
```
1.8.10

shubhamsinghmutualmobile commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
+#Thu Feb 16 04:11:02 IST 2023
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
```
Latest stable version is 7.6

shubhamsinghmutualmobile commented on settings.gradle:
```diff
@@ -1,8 +1,8 @@
 pluginManagement {
     plugins {
-        id("com.google.devtools.ksp") version "1.6.10-1.0.4"
+        id("com.google.devtools.ksp") version "1.7.21-1.0.8"
```
1.8.10-1.0.9

shubhamsinghmutualmobile commented on settings.gradle:
```diff
@@ -1,8 +1,8 @@
+        id("com.google.devtools.ksp") version "1.7.21-1.0.8"
         id 'org.jetbrains.kotlin.jvm'
-        id 'org.jetbrains.kotlin.android' version '1.5.21'
+        id 'org.jetbrains.kotlin.android' version '1.7.21'
```
1.8.10

shubhamsinghmutualmobile reviewed (changes requested):
Let's remove files coming in from `.idea`? And also, would it be a good time to introduce either `buildSrc` or composite builds to this project?

pushpalroymm committed:
Bump gradle version

pushpalroymm committed:
Bump kotlin, ksp, gradle, agp

pushpalroymm committed:
Use scaffold padding

pushpalroymm committed:
Replace deprecated onBackPressed

pushpalroymm committed:
Update .gitignore

pushpalroymm committed:
Remove unnecessary cache files

pushpalroymm commented:
> Let's remove files coming in from `.idea`? And also, would it be a good time to introduce either `buildSrc` or composite builds to this project?

Thanks for pointing it out @shubhamsinghmutualmobile.
I have removed the cached files and added them in .gitignore.

Sure we can work on buildSrc/composite build in a separate PR. 👍

pushpalroymm committed:
Bump ksp of Barricade compiler.

shubhamsinghmutualmobile reviewed (approved):
LGTM
